pkg/tfgen/edit_rules.go

```diff
@@ -29,24 +30,52 @@ func defaultEditRules() editRules {
 		boundedReplace("[tT]erraform [pP]lan", "pulumi preview"),
 		// Replace content such as " Terraform Apply." with " pulumi up."
 		boundedReplace("[tT]erraform [aA]pply", "pulumi up"),
-		reReplace(`"([mM])ade (by|with) [tT]erraform"`, `"Made $2 Pulumi"`),
+		reReplace(`"([mM])ade (by|with) [tT]erraform"`, `"Made $2 Pulumi"`, info.PreCodeTranslation),
 		// A markdown link that has terraform in the link component.
-		reReplace(`\[([^\]]*)\]\([^\)]*terraform([^\)]*)\)`, "$1"),
+		reReplace(`\[([^\]]*)\]\([^\)]*terraform([^\)]*)\)`, "$1", info.PreCodeTranslation),
 		fixupImports(),
 		// Replace content such as "jdoe@hashicorp.com" with "jdoe@example.com"
-		reReplace("@hashicorp.com", "@example.com"),
+		reReplace("@hashicorp.com", "@example.com", info.PreCodeTranslation),
+
+		// The following edit rules may be applied after translating the code sections in a document.
+		// Its primary use case is for the docs translation approach spearheaded in installation_docs.go.
+		// These edit rules allow us to safely transform certain strings that we would otherwise need in the
+		// code translation or nested type discovery process.
+		// These rules are currently only called when generating installation docs.
```

Comment:
Can we create a TODO here and link it to an issue.

```suggestion
		//
		// These rules are currently only called when generating installation docs. 
		// TODO[<issue here>]: Call info.PostCodeTranslation rules on all docs.
```